src/WhenThenGraphQL.js

```diff
@@ -0,0 +1,60 @@
+const isEqual = require("lodash.isequal");
+
+const contains = (obj, source) =>
+  Object.keys(source).every(
+    (key) => obj.hasOwnProperty(key) && obj[key] === source[key]
+  );
+
+const whenThenGraphQL = (server, graphql) => {
+  const when = ({ action, actionName }) => {
+    let count = 0;
+    let resolvers = [];
+
+    const thenReturn = ({responseBody, errors=[]}) => {
+      const resolver = (req, res, context) => {
+        return (errors.length > 0) 
+          ? res(context.errors(errors))
+          : res(context.data(responseBody));
+      }
+      resolvers.push(resolver);
+      return thenFunctions;
+    };
+
+    const thenReturnFor = (request, {responseBody, errors=[]}) => {
+      const {
+        headers: requestHeaders,
+      } = request;
+
+      const resolver = (req, res, context) =>
+        (!requestHeaders || contains(req.headers.map, requestHeaders)) &&
```

Comment:
For the rest-client we check against the body too, is there an equivalent we can do with graphql? Not sure if checking the correct query is being sent for example would be valuable here too?

Reply:
I'm not a hardcore `graphql` user neither and I think that the added features (just checking `action names` and `headers`) will cover 95% of the needs. 

If someone asks for it, I think that I could add an extra handler for something based on schemas: 
https://mswjs.io/docs/api/graphql/operation#resolve-against-schema

But I would suggest to start with a simpler API and wait to see if someone needs more.